conda/reporters.py

```diff
@@ -6,27 +6,60 @@
 
 from __future__ import annotations
 
+import logging
+import sys
+
+from .base.constants import (
+    DEFAULT_CONSOLE_REPORTER_BACKEND,
+    DEFAULT_JSON_REPORTER_BACKEND,
+)
 from .base.context import context
+from .exceptions import CondaError
+
+logger = logging.getLogger(__name__)
 
 
 def render(data, style: str | None = None, **kwargs) -> None:
-    for settings in context.reporters:
-        reporter = context.plugin_manager.get_reporter_backend(settings.get("backend"))
-        output = context.plugin_manager.get_reporter_output(settings.get("output"))
+    """
+    Used to render output in conda
+
+    The output will either be rendered as "json" or normal "console" output to stdout.
+    This function allows us to configure different reporter backends for these two types
+    of output.
+    """
+    if context.json is True:
+        backend = DEFAULT_JSON_REPORTER_BACKEND
+    elif isinstance(context.json, str):
+        backend = context.json
+    else:
+        backend = context.console
+
+    reporter = context.plugin_manager.get_reporter_backend(backend)
+
+    if reporter is None:
+        logger.warning(
+            f'Unable to find reporter backend: "{backend}"; '
+            f'falling back to using "{DEFAULT_CONSOLE_REPORTER_BACKEND}"'
+        )
+        reporter = context.plugin_manager.get_reporter_backend(
+            DEFAULT_CONSOLE_REPORTER_BACKEND
+        )
 
-        if reporter is None or output is None:
-            continue
+        if reporter is None:
+            raise CondaError(
+                "There are no available reporter backends to render output. This conda installation"
+                " is most likely corrupt and requires a re-installation."
+            )
```

Comment:
What about moving both of these checks into `get_reporter_backend` instead and simplify this into:

```python
reporter = context.plugin_manager.get_reporter_backend(backend, default=DEFAULT_CONSOLE_REPORTER_BACKEND)
```